From 5ae8c0323ea39a668c2946e0ef1c29efd88f9bec Mon Sep 17 00:00:00 2001
From: Ryan QIAN <b32804@freescale.com>
Date: Mon, 13 Feb 2012 13:41:29 +0800
Subject: [PATCH 1020/2463] ENGR00174311: [MX6Q]USDHC: SD3.0 card failed to
 work

- add auto_tune back on pretuning to improve compatibility.

Signed-off-by: Ryan QIAN <b32804@freescale.com>
---
 drivers/mmc/host/sdhci-esdhc-imx.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/mmc/host/sdhci-esdhc-imx.c b/drivers/mmc/host/sdhci-esdhc-imx.c
index af2c7fc..90c4b20 100644
--- a/drivers/mmc/host/sdhci-esdhc-imx.c
+++ b/drivers/mmc/host/sdhci-esdhc-imx.c
@@ -260,6 +260,7 @@ void esdhc_prepare_tuning(struct sdhci_host *host, u32 val)
 	reg = readl(host->ioaddr + SDHCI_MIX_CTRL);
 	reg |= SDHCI_MIX_CTRL_EXE_TUNE | \
 		SDHCI_MIX_CTRL_SMPCLK_SEL | \
+		SDHCI_MIX_CTRL_AUTO_TUNE | \
 		SDHCI_MIX_CTRL_FBCLK_SEL;
 	writel(reg, host->ioaddr + SDHCI_MIX_CTRL);
 	writel((val << 8), host->ioaddr + SDHCI_TUNE_CTRL_STATUS);
-- 
1.7.7.4

